editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/SearchResultCard/getDisplayDetailsForResult.tsx

```diff
@@ -0,0 +1,277 @@
+import {
+  ComponentType,
+  flatFlags,
+  IndexedNode,
+} from "@opensystemslab/planx-core/types";
+import { Calculate } from "@planx/components/Calculate/model";
+import { Confirmation } from "@planx/components/Confirmation/model";
+import { FileUploadAndLabel } from "@planx/components/FileUploadAndLabel/model";
+import { List } from "@planx/components/List/model";
+import { NextSteps } from "@planx/components/NextSteps/model";
+import { ChecklistField } from "@planx/components/shared/Schema/model";
+import { TaskList } from "@planx/components/TaskList/model";
+import { SearchResult } from "hooks/useSearch";
+import { capitalize, get } from "lodash";
+import { SLUGS } from "pages/FlowEditor/data/types";
+import { useStore } from "pages/FlowEditor/lib/store";
+
+/**
+ * Functions to map a search result to the fields required by SearchResultCard
+ */
+interface SearchResultFormatter {
+  getDisplayKey: (result: SearchResult<IndexedNode>) => string;
+  getIconKey: (result: SearchResult<IndexedNode>) => ComponentType;
+  getTitle: (result: SearchResult<IndexedNode>) => string;
+  getHeadline: (result: SearchResult<IndexedNode>) => string;
+  getComponentType: (result: SearchResult<IndexedNode>) => string;
+}
+
+type KeyMap = Record<string, Partial<SearchResultFormatter>>;
+
+type ComponentMap = Partial<
+  Record<ComponentType, Partial<SearchResultFormatter>>
+>;
+
+const keyFormatters: KeyMap = {
+  "data.fn": {
+    getDisplayKey: () => "Data",
+  },
+  "data.val": {
+    getDisplayKey: ({ item }) =>
+      item.type === ComponentType.Answer ? "Option (data)" : "Option",
+  },
+  "data.fileTypes.fn": {
+    getDisplayKey: () => "File type (data)",
+    getHeadline: ({ item, refIndex }) =>
+      (item.data as unknown as FileUploadAndLabel).fileTypes[refIndex].fn!,
+  },
+  "data.dataFieldBoundary": {
+    getDisplayKey: () => "Field boundary",
```

Comment:
nit: maybe `Draw boundary (data)`  & `Draw boundary (area)` are more clear names for these two? "Field" can feel like a sort of loaded generic word to me!